lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -44,11 +45,18 @@ def build_apps_v1beta1_kubeclient(context)
     end
 
     def _build_kubeclient(api_version:, context:, endpoint_path: nil)
-      config = GoogleFriendlyConfig.read(ENV.fetch("KUBECONFIG"))
-      unless config.contexts.include?(context)
+      # Find a context defined in kube conf files that matches the input context by name
+      kube_context = nil
+      config_files.each do |f|
```

Comment:
This works, but in Ruby a more idiomatic approach may be:

```ruby
friendly_configs = config_files.map { |f| GoogleFriendlyConfig.read(f) }
config = friendly_configs.find { |c| c.contexts.include?(context) }

raise ContextMissingError, context unless config

kube_context = config.context(context)
```

(You could get fancier with lazy enumerators and the like.)

Reply:
Good point. Added lazy loading.